lib/matplotlib/axes/_base.py

```diff
@@ -2858,14 +2858,9 @@ def ticklabel_format(self, *, axis='both', style='', scilimits=None,
                 m + n + 1  # check that both are numbers
             except (ValueError, TypeError):
                 raise ValueError("scilimits must be a sequence of 2 integers")
-        if style[:3] == 'sci':
-            sb = True
-        elif style == 'plain':
-            sb = False
-        elif style == '':
-            sb = None
-        else:
-            raise ValueError("%s is not a valid style value")
+        STYLES = {'sci': True, 'scientific': True, 'plain': False, '': None}
+        cbook._check_in_list(STYLES, style=style)
+        sb = STYLES[style]
```

Comment:
not to derail this PR, but I had some thought of adding a `_check_in_dict` such that one could write e.g. here
```
sb = cbook._check_in_dict(
    {'sci': True, 'scientific': True, 'plain': False, '': None}, style=style)
```
as the pattern is quite common throughout the codebase; or even fold the functionality into _check_in_list (or further bikeshed the name :p)
Thoughts?

Reply:
That's not quite the right name then `check_in_dict` does returning any value (except maybe bool). Also, it's not clear that this does a lookup.

Either we name the function differently, e.g.`_get_from_dict_checked`,  `_get_checked`, ....

or we make it a more general_patttern `parse_parameter(validator, param)` where `validator` could be a list, a dict or a function`,

or we do a `ParameterDict` with the respective message on `__getitem__` failure:
~~~
STYLES = ParameterDict(sci=True, scientific=True, plain=False)
sb = STYLES[style]
~~~

Reply:
That's why I said "or further bikeshed" :)

Reply:
And I'm putting my sheds next to it :).

Reply:
I guess _check_get would go well with _check_in_list, _check_isinstance?

Reply:
Maybe ` _check_getitem`? My concern with `_check_get` is that `dict.get` returns None or a default, for nonexisting items whereas `_check_get` will raise.

Reply:
Sounds good.  Wanna PR it? :p

Reply:
Will do later :p

Reply:
now done :)